File: frontend/src/types/order.ts
export interface Order {
  id: number
  orderNumber: string
  product: Product
  buyer: User
  seller: User
  price: number
  finalPrice: number
  status: OrderStatus
  contactInfo: string
  deliveryAddress: string
  deliveryMethod: string
  paymentMethod: string
  remarks?: string
  createdAt: string
  updatedAt: string
}

export enum OrderStatus {
  PENDING = 'PENDING',
  CONFIRMED = 'CONFIRMED',
  SHIPPED = 'SHIPPED',
  DELIVERED = 'DELIVERED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  REFUNDED = 'REFUNDED'
}

export interface OrderCreateDTO {
  productId: number
  price: number
  contactInfo: string
  deliveryAddress: string
  deliveryMethod: string
  paymentMethod: string
  remarks?: string
}

export interface OrderUpdateDTO {
  contactInfo?: string
  deliveryAddress?: string
  deliveryMethod?: string
  paymentMethod?: string
  remarks?: string
}

export interface OrderStatistics {
  totalOrders: number
  pendingOrders: number
  confirmedOrders: number
  completedOrders: number
  cancelledOrders: number
  totalAmount: number
  monthlyOrders: number
  monthlyAmount: number
}

// 导入其他相关类型
import { Product } from './product'
import { User } from './user'

File: backend/src/main/java/com/campus/secondhandplatform/entity/Order.java
package com.campus.secondhandplatform.entity;

import javax.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
@Data
@EqualsAndHashCode(callSuper = false)
@EntityListeners(AuditingEntityListener.class)
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_number", nullable = false, unique = true)
    private String orderNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer_id", nullable = false)
    private User buyer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(name = "final_price")
    private BigDecimal finalPrice; // 最终成交价

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status = OrderStatus.PENDING;

    @Column(name = "contact_info")
    private String contactInfo;

    @Column(name = "delivery_address")
    private String deliveryAddress;

    @Column(name = "delivery_method")
    private String deliveryMethod;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "payment_status")
    private String paymentStatus = "UNPAID";

    @Column(columnDefinition = "TEXT")
    private String remarks; // 备注

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum OrderStatus {
        PENDING, CONFIRMED, PAID, SHIPPED, COMPLETED, CANCELLED, REFUNDED
    }
}